lib/src/models/metric_value.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:meta/meta.dart';
+
+import 'context_message.dart';
+import 'metric_documentation.dart';
+import 'metric_value_level.dart';
+
+/// Represents a value computed by the metric
+@immutable
+class MetricValue<T> {
+  /// The id of the metric whose compute this value
+  final String metricsId;
+
+  /// Documentation associated with the metrics whose compute this value
+  final MetricDocumentation documentation;
+
+  /// The actual value computed by the metric
+  final T value;
+
+  /// Level of this value computed by the metric
+  final MetricValueLevel level;
+
+  /// Message for user containing information about this value
+  final String comment;
+
+  /// Message for user containing information about how the user can improve this value
+  final String recommendation;
```

Comment:
Correction: `The message for the user containing information about how the user can improve this value`